mayastor/src/lvs/lvol.rs

```diff
@@ -330,4 +352,55 @@ impl Lvol {
             }
         }
     }
+
+    /// Format snapshot name
+    /// base_name is the nexus or replica UUID
+    pub fn format_snapshot_name(base_name: &str, snapshot_time: u64) -> String {
+        format!("{}-snap-{}", base_name, snapshot_time)
+    }
+
+    /// Create a snapshot
+    pub async fn create_snapshot(
+        &self,
+        nvmf_req: &NvmfReq,
+        snapshot_name: &str,
+    ) {
+        extern "C" fn snapshot_done_cb(
+            nvmf_req_ptr: *mut c_void,
+            _lvol_ptr: *mut spdk_lvol,
+            errno: i32,
+        ) {
+            let nvmf_req = NvmfReq(
```

Comment:
If you do a `Impl From<T> for NvmfRequest {}` the NonNull::new().unwrap() part can be handled there.

Reply:
Thanks, fixed.